src/browser/provider/built-in/chrome.js

```diff
@@ -0,0 +1,384 @@
+}
+
+function buildChromeArgs (config, cdpPort, platformArgs, userDataDir) {
+    return [`--remote-debugging-port=${cdpPort}`, `--user-data-dir=${userDataDir.name}`]
+        .concat(
+            config.headless ? ['--headless'] : [],
+            config.userArgs ? [config.userArgs] : [],
+            platformArgs ? [platformArgs] : []
+        )
+        .join(' ');
+}
+
+async function killChrome (cdpPort) {
+    var chromeOptions = { arguments: `--remote-debugging-port=${cdpPort}` };
+    var chromeProcess = await psLookup(chromeOptions);
+
+    if (!chromeProcess.length)
+        return true;
+
+    try {
+        await psKill(chromeProcess[0].pid, { timeout: BROWSER_CLOSING_TIMEOUT });
+
+        return true;
+    }
+    catch (e) {
+        return false;
+    }
+}
+
+async function stopLocalChrome (cdpPort) {
+    if (!await killChrome(cdpPort))
+        await killChrome(cdpPort);
+}
+
+async function getActiveTab (cdpPort, browserId) {
+    var tabs = await remoteChrome.listTabs({ port: cdpPort });
+    var tab  = tabs.filter(t => t.type === 'page' && t.url.indexOf(browserId) > -1)[0];
+
+    return tab;
+}
+
+async function setEmulationBounds (client, device, overrideWidth, overrideHeight) {
+    var width  = overrideWidth !== void 0 ? overrideWidth : device.width;
+    var height = overrideHeight !== void 0 ? overrideHeight : device.height;
+
+    await client.Emulation.setDeviceMetricsOverride({
+        width:             width,
+        height:            height,
+        deviceScaleFactor: device.density,
+        mobile:            device.mobile,
+        fitWindow:         true
+    });
+
+    await client.Emulation.setVisibleSize({ width, height });
+}
+
+async function setEmulation (client, device) {
+    if (device.userAgent !== void 0)
+        await client.Network.setUserAgentOverride({ userAgent: device.userAgent });
+
+    if (device.touch !== void 0) {
+        await client.Emulation.setTouchEmulationEnabled({
+            enabled:       device.touch,
+            configuration: device.mobile ? 'mobile' : 'desktop'
+        });
+    }
+
+    await setEmulationBounds(client, device);
+}
+
+async function getWindowId (client, tab) {
+    try {
+        var { windowId } = await client.Browser.getWindowForTarget({ targetId: tab.id });
+
+        return windowId;
+    }
+    catch (e) {
+        return null;
+    }
+}
+
+async function getCdpClientInfo (cdpPort, browserId) {
+    try {
+        var tab = await getActiveTab(cdpPort, browserId);
+
+        if (!tab)
+            return {};
+
+        var client   = await remoteChrome({ target: tab, port: cdpPort });
+        var windowId = await getWindowId(client, tab);
+
+        return { tab, client, windowId };
+    }
+    catch (e) {
+        return {};
+    }
+}
+
+function createTempUserDataDir () {
+    tmp.setGracefulCleanup();
+
+    return tmp.dirSync({ unsafeCleanup: true });
+}
+
+export default {
+    openedBrowsers: {},
```

Comment:
the file is huge. My proposal:
`built-in/chrome/index.js` - with provider definition
and several logical units in `built-in/chrome` with other functions